### PR TITLE
feat: SLA threshold card component

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

This PR includes
1. Component to display SLA timer in the conversation card and header
2. Resize directive that I took from this [commit](https://github.com/chatwoot/chatwoot/pull/9109/commits/af90e3a56cff138db408a698d1a5a21b69072263) by @scmmishra 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

**Screenshots**
Screenshot 2024-03-27 at 12 02 24 PM
Screenshot 2024-03-27 at 12 07 49 PM
Screenshot 2024-03-27 at 12 07 54 PM
Screenshot 2024-03-27 at 12 07 59 PM
Screenshot 2024-03-27 at 12 09 12 PM
Screenshot 2024-03-27 at 12 08 02 PM

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/components/widgets/conversation/components/SLACardLabel.vue [hunk 1]
```diff
@@ -0,0 +1,123 @@
+<template>
+  <div
+    v-if="showSLACard"
```
Comment (scmmishra) [inline app/javascript/dashboard/components/widgets/conversation/components/SLACardLabel.vue:3]: Top level v-if's are not a good idea. The component should never decide if they will toggle themselves, best is to let the parent handle this. That way we won't have to pass that additional prop `showSLACard`

File: app/javascript/dashboard/components/widgets/conversation/components/SLACardLabel.vue [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    class="flex items-center px-2 truncate border border-solid min-w-fit w-fit border-slate-75 dark:border-slate-700"
```
Comment (scmmishra) [inline app/javascript/dashboard/components/widgets/conversation/components/SLACardLabel.vue:4]:
```suggestion
    class="flex items-center px-2 truncate border w-fit border-slate-75 dark:border-slate-700"
```

- border by default is solid
- `min-w-fit` has the same effect as `w-fit`, we would need only one of those

File: app/javascript/dashboard/components/widgets/conversation/components/SLACardLabel.vue [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    :class="
+      showExtendedInfo
+        ? 'py-[5px] h-[26px] rounded-lg'
```
Comment (scmmishra) [inline app/javascript/dashboard/components/widgets/conversation/components/SLACardLabel.vue:7]: If we have h-[26px] will we need `py-[5px]`?

File: app/javascript/dashboard/components/widgets/conversation/components/SLACardLabel.vue [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    this.createTimer();
+  },
+  beforeDestroy() {
+    if (this.timer) {
+      clearTimeout(this.timer);
+    }
+  },
+  methods: {
+    createTimer() {
+      this.timer = setTimeout(() => {
```
Comment (scmmishra) [inline app/javascript/dashboard/components/widgets/conversation/components/SLACardLabel.vue:113]: One concern I have with this is that there is an SLA timer for every instance of `SLACardLabel`. We could figure out a way to have a centralized timer.

Can you try to see if this causes any perf issues when you have ~200 conversations in the list?

For now, this is good to merge.

**Review by scmmishra — CHANGES_REQUESTED** [review]

File: app/javascript/dashboard/components/widgets/conversation/components/SLACardLabel.vue (hunk 4 above)
Comment (iamsivin) [reply app/javascript/dashboard/components/widgets/conversation/components/SLACardLabel.vue]: @scmmishra I have commented these changes from this PR. I will enable this PR https://github.com/chatwoot/chatwoot/pull/9146 and will find a way to centralize the timer.

**Review by scmmishra — APPROVED** [review]

**Pull request merged**